spotpy/unittests/test_parameter.py

```diff
@@ -280,5 +279,119 @@ def test_triangular_has_correct_statistics(self):
         self.assertAlmostEqual(np.std(nums), expected_sd, self.tolerance, "SD of Triangular({}, {}, {}) should be {}".format(self.a, self.c, self.b, expected_sd))
 
 
+class TestParameterArguments(unittest.TestCase):
+    """
+    Test by philippkraft
+    checks behaviour of Uniform and Triangular for a number of
+    different arguments.
+    """
+
+    def setUp(self):
+        """
+        Setup for 2 simple parameter cases
+        :return:
+        """
+        self.classes = [parameter.Uniform, parameter.Triangular]
+        self.rndargs = [(10, 20), (10, 15, 20)]
+
+    def test_correct_with_default_step(self):
+        for cl, args in zip(self.classes, self.rndargs):
+            p_no_name = cl(*args)
+            p_with_name = cl(cl.__name__, *args)
+            self.assertTrue(10 <= p_no_name.optguess < 20, 'Optguess out of boundaries')
+            self.assertTrue(10 <= p_with_name.optguess < 20, 'Optguess out of boundaries')
+            self.assertTrue(p_no_name.step < 10, 'Step to large')
+            self.assertTrue(p_with_name.step < 10 , 'Step to large')
+
+    def test_correct_with_extra_args(self):
+        for cl, args in zip(self.classes, self.rndargs):
+            p_no_name = cl(*args, step=1, default=12)
+            p_with_name = cl(cl.__name__, *args, step=1, default=12)
+            self.assertTrue(p_no_name.optguess == 12,
+                            'Optguess not found from default (name={})'.format(repr(p_no_name.name)))
+            self.assertTrue(p_with_name.optguess == 12,
+                            'Optguess not found from default (name={})'.format(repr(p_with_name.name)))
+            self.assertTrue(p_no_name.step == 1,
+                            'Step overridden by class (name={})'.format(repr(p_no_name.name)))
+            self.assertTrue(p_with_name.step == 1,
+                            'Step overridden by class (name={})'.format(repr(p_with_name.name)))
+
+def make_args(pcls):
+    """
+    Returns an args tuple for the parameter class pcls.
+    """
+    return tuple(range(1, len(pcls.__rndargs__) + 1))
+
+
+class TestParameterClasses(unittest.TestCase):
+    """
+    Test by philippkraft to test all available Parameter classes, except List
+    """
+
+    def setUp(self):
+        """
+        Get all classes from spotpy.parameter module, except special cases
+        """
+        self.classes = []
+        for cname, cls in vars(parameter).items():
+            if (cls is type
+                    and cls not in [parameter.Base, parameter.List]
+                    and issubclass(cls, parameter.Base)):
+                self.classes.append(cls)
+
+
+    def test_create_posargs(self):
+        """
+        Checks if the right number of arguments is present
+        :return:
+        """
+        for cname, cls in self.classes:
+            # Add step parameter to args
+            args = make_args(cls) + (0.01,)
+            p = cls(cname, *args)
+            self.assertFalse(p.name)
```

Comment:
I don't understand this fully, could you please specify why exactly are we asserting False on `p.name`? Is it not supposed to be initialized at this stage?

Reply:
I don't either :-). The test should have failed, but the setUp method has not been called. I am going to update the test.